LeadKitAdditions/Sources/Services/ValidationService/ValidationItem.swift

```diff
@@ -0,0 +1,98 @@
+import SwiftValidator
+import RxSwift
+import RxCocoa
+
+enum ValidationItemState {
+    case initial
+    case correction(ValidationError)
+    case error(ValidationError)
+    case valid
+}
+
+extension ValidationItemState {
+
+    var isInitial: Bool {
+        switch self {
+        case .initial:
+            return true
+        default:
+            return false
+        }
+    }
+
+    var isValid: Bool {
+        switch self {
+        case .valid:
+            return true
+        default:
+            return false
+        }
+    }
+
+}
+
+class ValidationItem {
+
+    private let disposeBag = DisposeBag()
+
+    private let validationStateHolder: Variable<ValidationItemState> = Variable(.initial)
+    var validationState: ValidationItemState {
+        return validationStateHolder.value
+    }
+    var validationStateObservable: Observable<ValidationItemState> {
+        return validationStateHolder.asObservable()
+    }
+
+    private(set) var rules: [Rule] = []
+    private var text: String?
+
+    init(textObservable: Observable<String?>, rules: [Rule]) {
+        self.rules = rules
+        bindValue(with: textObservable)
+    }
+
+    private func bindValue(with textObservable: Observable<String?>) {
+        textObservable
+            .do(onNext: { [weak self] value in
+                self?.text = value
+            })
+            .filter { [weak self] _ in !(self?.validationState.isInitial ?? true)}
+            .subscribe(onNext: { [weak self] value in
+                self?.validate(text: value)
+            })
+            .addDisposableTo(disposeBag)
+    }
+
+    @discardableResult
+    func manualValidate() -> Bool {
+        return validate(text: text, isManual: true)
+    }
+
+    @discardableResult
```

Comment:
I believe it do not stop here. I'd like to give one mo way to get validation result